app/models/domain_response.rb

```diff
@@ -0,0 +1,99 @@
+# A stateless response to a given set of questions in one domain.
+# Initializing is done by setting answer_values as a hash, for example:
+#   r = DomainResponse.new(domain_id: 'POS-PQ', answer_values: { 'EL02' => 1 })
+#
+# After this, all attributes can be read in an OOP way, for example:
+#   r.questions # All questions, including the next (unanswered) one
+#   # => [#<Question:0x007faadb3459a8 @id="EL02", @answer_value: 1>,
+#         #<Question:0x007faadb2d49d8 @id="EL03">]
+class DomainResponse < BaseModel
+  # @see https://github.com/roqua/screensmart-r/blob/master/inst/extdata/estimate_interpretations_quartiles.csv
+  QUARTILE_MAPPINGS = {
+    'Laag niveau (++)'            => 'Q1', # Psychopathologie
+    'Matig niveau (+)'            => 'Q2',
+    'Verhoogd niveau (-)'         => 'Q3',
+    'Sterk verhoogd niveau (--)'  => 'Q4',
+    'Laag niveau (--)'            => 'Q1', # Positieve constructen
+    'Matig niveau (-)'            => 'Q2',
+    'Hoog niveau (+)'             => 'Q3',
+    'Zeer hoog niveau (++)'       => 'Q4'
+  }.freeze
+
+  DOMAIN_SIGNS = {
+    'POS-PQ'    => 'neg',
+    'NEG-PQ'    => 'neg',
+    'ANX-PRO'   => 'neg',
+    'DEP-PRO'   => 'neg',
+    'SAT-PRO'   => 'pos',
+    'FRS-PRO'   => 'pos',
+    'EMO-PRO'   => 'pos',
+    'DIS-4DKL'  => 'neg'
+  }.freeze
+
+  NORM_POPULATION_LABELS = {
+    'pos' => 'Algemene bevolking',
+    'neg' => 'Cliënten eerste lijn GGZ'
+  }.freeze
+
+  attr_accessor :uuid, :domain_id
+
+  # accessors for attributes defined by R package
+  %i(next_question_id estimate variance done estimate_interpretation warning).each do |r_attribute|
+    define_method r_attribute do
+      ensure_valid do
+        RPackage.data_for(answer_values, [domain_id])[r_attribute]
+      end
+    end
+  end
+
+  def questions
+    next_question.present? ? completed_questions.push(next_question) : completed_questions
```

Comment:
Zelf denk ik vaak in sets, op die manier zou dit ook een mooie manier kunnen zijn waar geen if conditie voor nodig is:
`completed_questions + next_questions`

next_questions
```
def next_questions
  done ? [] : [Question.new id: next_question_id, domain_id: domain_id]
end
```

Het is maar een idee, wat er nu staat is niet verkeerd.